### DISCUSSION TITLE
I try to use the index of Clangd to generate it in memory, but it doesn't work

### DISCUSSION THREAD

**Discussion by Hebe970217** [clangd/clangd#1046] [Q&A]
ps:
`/usr/lib/x86_64-linux-gnu/unioncode/tools/clangd -log=verbose --clang-tidy --pch-storage=memory -j=4 --header-insertion=never --header-insertion-decorators --all-scopes-completion`

show clangd help
`--background-index              - Index project code in the background and persist index on disk.`
where is index file generated when I unuse "--background-index" ? memory or disk?

**Comment by HighCommander4** [comment]
Without `--background-index`, there is indeed an in-memory index, but only opened files are added to it (whereas `--background-index` will look at `compile_commands.json` and index the files listed there).

**Reply by Hebe970217** [reply]
Excuse me, How to do with generate index in memory? I don't want to generate the index to disk.

**Reply by Hebe970217** [reply]
When I use Clangd, I don't want to have the whole .cache folder

**Reply by Hebe970217** [reply]
Is there any good solution?

**Reply by HighCommander4** [reply]
Maybe with a RAM disk (`tmpfs` on Linux, see e.g. https://www.techrepublic.com/article/how-to-use-a-ramdisk-on-linux/). You can copy your `compile_commands.json` onto the RAM disk (this will determine the location of the `.cache` directory), and point clangd to it using `--compile-commands-dir`.
